compiler/optimizer/DeadTreesElimination.cpp

```diff
@@ -480,20 +485,37 @@ int32_t TR::DeadTreesElimination::performOnBlock(TR::Block *block)
    return 0;
    }
 
+typedef std::pair<TR::Node* const, TR::TreeTop* const> ReadBarToTreeTopMapEntry;
+typedef TR::typed_allocator<ReadBarToTreeTopMapEntry, TR::Region&> ReadBarToTreeTopMapAlloc;
+typedef std::map<TR::Node*, TR::TreeTop *, std::less<TR::Node*>, ReadBarToTreeTopMapAlloc> ReadBarToTreeTopMap;
```

Comment:
This is going to need some memory profililng - std::map can be very inefficient. Alternative could be std::deque where we index using the node global index to look up a TreeTop or an array sized for the max node global index. Profiling the memory footprint increase is certainly something that needs to be done along with the high water mark.